### PR TITLE
MEM-52: Updated flask routes with proper versioning

### PR DESCRIPTION
**Pull request by jjangsangy**

### CODE REVIEW

File: src/environment/common.cfg.j2 [hunk 1]
```diff
@@ -22,4 +22,5 @@ endpoints           = ["twitter", "instagram", "youtube", "phone", "geocode"]
 [api]
 username           = "{{ credstash('flask_api_username') }}"
 password           = "{{ credstash('flask_api_password') }}"
+version            = "{{ credstash('flask_api_version') }}"
```
Comment (echu) [inline src/environment/common.cfg.j2]: Does this have to be a secret? We could just put "v1" or "v2" in there, right?

File: src/server.py [hunk 2]
```diff
@@ -25,9 +25,11 @@
 from flask.ext.restful   import Api
 
 app  = Flask(__name__)
-cors = CORS(app, resources={r"/enhance/*": {"origins": "*"}, r"/search":    {'origins': '*'}})
+cors = CORS(app)
```
Comment (echu) [inline src/server.py]: I'm not sure what the CORS stuff does, but does removing the `resources` argument break anything?

**Comment by echu** [comment]
Minor questions I need answered, but otherwise, pretty straightforward.

File: src/server.py (hunk 2 above)
Comment (jjangsangy) [inline src/server.py]: `resource={}` is a dict that allows you to specify exactly which endpoints and values to set the endpoint.

In some situations, you would like to only allow cross-origin access to certain resources, from only certain IP addresses. If you leave out the `resources` what happens is that it opens access to all endpoints with `*`, which I think is deserving of a lol

File: src/environment/common.cfg.j2 (hunk 1 above)
Comment (jjangsangy) [inline src/environment/common.cfg.j2]: Good point, I can go make that change and then remove the version from credstash?

File: src/server.py (hunk 2 above)
Comment (echu) [inline src/server.py]: Sounds good. Maybe leave a comment on top of line 28 stating something to this effect (so someone else down the line isn't surprised).

I'm guessing that we'll need to restrict it to swaggerhub.com in "production", so maybe we start preparing for that now.

**Pull request merged**